pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -222,9 +222,32 @@ func (pc *PodCache) getPodByIP(addr string) *v1.Pod {
 	if !exists {
 		return nil
 	}
-	item, exists, err := pc.informer.GetStore().GetByKey(key)
-	if !exists || err != nil {
-		return nil
+	return pc.getPodByKey(key)
+}
+
+// getPodByKey returns the pod or nil if pod not found or an error occurred
+func (pc *PodCache) getPodByKey(key string) *v1.Pod {
```

Comment:
nit: would be nice to add a comment about what the key format is

Reply:
make sense